### ISSUE TITLE
StreamingBody/contentLength uses .available wrongly

### ISSUE THREAD

**Issue by leahneukirchen** [into-docker/unixsocket-http#15]
`-contentLength` returns `.available`, if positive, as the Content-Length of the body.

However, `InputStream` only is only specified to *Returns an **estimate** of the number of bytes that can be read (or skipped over) from this input stream without blocking by the next invocation of a method for this input stream.* In particular, it is an estimate, and only applies for a single read. Therefore, all uses of `StreamingBody` where multiple reads are required (e.g. a `PipedInputStream`) or where it's only an estimate result in errors like this:

```
Unhandled java.net.ProtocolException:  expected 1024 bytes but received 2048
               Exchange.kt:  214  okhttp3.internal.connection.Exchange$RequestBodySink/write
       RealBufferedSink.kt:  279  okio.RealBufferedSink/close
...
```

I think the InputStream interface is too weak to guarantee correct total size, and chunked transfer should always be used when having an InputStream as a body.

**Comment by xsc** [comment]
@leahneukirchen Thanks! You're absolutely right. There might have been a reason I implemented it like this (e.g. specific Docker API endpoints explicitly expecting a `Content-Length` header) but I'm not sure. Also, if that's the case, why didn't I add a comment?

Anyway, I'll check it out.

**Comment by leahneukirchen** [comment]
At least for `PutContainerArchive`, chunked works fine.

**Comment by xsc** [comment]
Released as `[unixsocket-http "1.0.11"]`. Thanks again!
